### PR TITLE
Docs: update glsl-generate docs and doc --skip-render

### PR DESCRIPTION
**Pull request by paulthomson**

### CODE REVIEW

File: docs/glsl-fuzz-walkthrough.md [hunk 1]
```diff
@@ -146,6 +146,9 @@ cp -r graphicsfuzz-1.0/shaders/samples samples
 # Synopsis:
 # glsl-generate [options] references donors num_variants glsl_version prefix output_folder
 
+# Add `--seed 12345` if you want to enure that running the same glsl-generate command again
```
[the remark this answers is not in the document]
Comment (paulthomson) [reply docs/glsl-fuzz-walkthrough.md]: I removed this.

File: docs/glsl-fuzz-walkthrough.md [hunk 2]
```diff
@@ -401,7 +404,9 @@ or you can install it using `adb`.
 There is no point in manually running this app from the Android device; it will crash unless
 it finds shaders in the `/sdcard/graphicsfuzz` directory.
 
-You can run the worker as follows:
+You can run the worker as follows.
+
+> Note that `glsl-to-spv-worker` assumes `adb` is on your PATH.
```
[the remark this answers is not in the document]
Comment (paulthomson) [reply docs/glsl-fuzz-walkthrough.md]: Done

File: docs/glsl-fuzz-reduce.md [hunk 3]
```diff
@@ -52,7 +63,14 @@ the following are common reduction commands:
...
 glsl-reduce shader-job.json --preserve-semantics --reduction-kind ABOVE_THRESHOLD --reference reference.info.json
 
-# No image reduction:
+# No image reduction for a crash that occurs during shader compilation.
+# --skip-render can be used since the error occurs during shader compilation;
+# successful shader compilation will be regarded as uninteresting. 
+glsl-reduce shader-job.json --reduction-kind NO_IMAGE --error-string "Fatal signal 11" --skip-render
+
+# No image reduction for a crash that occurs during rendering.
+# --skip-render should NOT be used since the error occurs during rendering;
```
[the remark this answers is not in the document]
Comment (paulthomson) [reply docs/glsl-fuzz-reduce.md]: Done

File: docs/glsl-fuzz-reduce.md [hunk 4]
```diff
@@ -52,7 +63,14 @@ the following are common reduction commands:
 # Wrong image reduction.
 glsl-reduce shader-job.json --preserve-semantics --reduction-kind ABOVE_THRESHOLD --reference reference.info.json
 
-# No image reduction:
+# No image reduction for a crash that occurs during shader compilation.
+# --skip-render can be used since the error occurs during shader compilation;
```
[the remark this answers is not in the document]
Comment (paulthomson) [reply docs/glsl-fuzz-reduce.md]: Done

File: docs/glsl-fuzz-reduce.md [hunk 5]
```diff
@@ -39,11 +39,22 @@ and the reference image will no longer be valid.
...
+  * This reduction kind is often used with `--skip-render`,
+which causes remote workers to skip rendering the shader.
+This can be much faster.
+In cases
+where the interesting error occurs during shader compilation,
+there is no need to render the shader to check if the error still occurs
+so `--skip-render` should be used.
+However, if the error occurs while rendering the shader,
+or as a result of rendering the shader,
+`--skip-render` should **not** be used.
```
[the remark this answers is not in the document]
Comment (paulthomson) [reply docs/glsl-fuzz-reduce.md]: Done

File: docs/glsl-fuzz-generate.md [hunk 6]
```diff
@@ -101,13 +128,13 @@ mkdir -p work/shaderfamilies
 # glsl-generate [options] donors references num_variants glsl_version prefix output_folder
 
 # Generate some GLSL version 300 es shaders.
-glsl-generate --max_bytes 500000 --multi_pass samples/donors samples/300es 10 "300 es" family_300es work/shaderfamilies
+glsl-generate samples/donors samples/300es 10 "300 es" family_300es work/shaderfamilies
```
[the remark this answers is not in the document]
Comment (paulthomson) [reply docs/glsl-fuzz-generate.md]: I added a note about changing the seed in the walk through, but not here.

File: docs/glsl-fuzz-walkthrough.md [hunk 7]
```diff
@@ -23,13 +23,13 @@ in this walkthrough.
 
 ## Requirements
 
-**Summary:** the latest release zip and worker applications, Java 8+, and Python 3.5+.
+**Summary:** the latest release zip and worker applications, Java 8+, Python 3.5+, and `adb` on your path.
```
Comment (afd) [inline docs/glsl-fuzz-walkthrough.md]: Possibly make it clear that adb is optional if you don't care about Android, e.g.:

"... Python 3.5+, and (if you want to do Android testing) `adb` on your path."

Comment (paulthomson) [reply docs/glsl-fuzz-walkthrough.md]: Good point. Done.

**Pull request merged**